app/controllers/bookmarks_controller.rb

```diff
@@ -0,0 +1,24 @@
+class BookmarksController < ApplicationController
+  def index
+    @bookmarks = Bookmark.where(user: current_user)
+
+  end
+
+  def create
+    @trail = Trail.find(params[:trail_id])
+    @bookmark = Bookmark.new(user: current_user, trail: @trail)
+
+    if @bookmark.save
+      redirect_to bookmarks_path, notice: "Trip is saved as bookmark!"
+    else
+      redirect_to trail_path(@trail), :alert => @bookmark.errors.full_messages[0]
+    end
+  end
+
+  def destroy
+    @bookmark.destroy
```

Comment:
You have to define @bookmark first
<img width="589" alt="Screenshot 2022-11-24 at 16 59 15" src="https://user-images.githubusercontent.com/109951226/203738677-1fa4ba59-0adb-40e6-8ccd-edac85cfa5b4.png">